java/org/apache/coyote/AbstractProtocol.java

```diff
@@ -769,7 +769,7 @@ private void logPortOffset() {
 
     // ------------------------------------------- Connection handler base class
 
-    protected static class ConnectionHandler<S> implements AbstractEndpoint.Handler<S> {
+    public static class ConnectionHandler<S> implements AbstractEndpoint.Handler<S> {
```

Comment:
Why do you need this change? Isn't the change to 'getHandler()' sufficient to expose the global request processor?

Reply:
Had the idea to make getGlobal returning RequestInfo directly (`public RequestGroupInfo getGlobal() { return global; }`) but didn't push it so you are right, will revert this change